openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+                else:
+                    # If not a user-requested Filter, get all bins
+                    if isinstance(self_filter, openmc.DistribcellFilter):
+                        # Create list of cell instance IDs for distribcell Filters
+                        bins = list(range(self_filter.num_bins))
+
+                    elif isinstance(self_filter, openmc.EnergyFunctionFilter):
+                        # EnergyFunctionFilters don't have bins so just add a None
+                        bins = [None]
+
+                    else:
+                        # Create list of IDs for bins for all other filter types
+                        bins = self_filter.bins
+
+                # Add indices for each bin in this Filter to the list
+                indices = np.array([self_filter.get_bin_index(b) for b in bins])
+                filter_indices.append(indices)
+
+                # Account for stride in each of the previous filters
+                for indices in filter_indices[:i]:
+                    indices *= self_filter.num_bins
+
+            # Apply outer product sum between all filter bin indices
+            filter_indices = list(map(sum, product(*filter_indices)))
+
+        # If user did not specify any specific Filters, use them all
+        else:
+            filter_indices = np.arange(self.num_filter_bins)
+
+        return filter_indices
+
+    def get_nuclide_indices(self, nuclides):
+        """Get indices into the nuclide axis of this tally's data arrays.
+
+        This is a helper method for the Tally.get_values(...) method to
+        extract tally data. This method returns the indices into the nuclide
+        axis of the tally's data array (axis=1) for one or more nuclides.
+
+        Parameters
+        ----------
+        nuclides : list of str
+            A list of nuclide name strings
+            (e.g., ['U235', 'U238']; default is [])
+
+        Returns
+        -------
+        numpy.ndarray
+            A NumPy array of the nuclide indices
+
+        """
+
+        cv.check_iterable_type('nuclides', nuclides, str)
+
+        # Determine the score indices from any of the requested scores
+        if nuclides:
+            nuclide_indices = np.zeros(len(nuclides), dtype=int)
+            for i, nuclide in enumerate(nuclides):
+                nuclide_indices[i] = self.get_nuclide_index(nuclide)
+
+        # If user did not specify any specific Nuclides, use them all
+        else:
+            nuclide_indices = np.arange(self.num_nuclides)
+
+        return nuclide_indices
+
+    def get_score_indices(self, scores):
+        """Get indices into the score axis of this tally's data arrays.
+
+        This is a helper method for the Tally.get_values(...) method to
+        extract tally data. This method returns the indices into the score
+        axis of the tally's data array (axis=2) for one or more scores.
+
+        Parameters
+        ----------
+        scores : list of str
```

Comment:
seems like you're expecting type of `(str, openmc.CrossScore)` in the type check. Let's update this documentation